File: pr1-java/server/src/main/java/pr1Java/server/reflection/ReflectionConcurrentServer.java
package pr1Java.server.reflection;

import pr1Java.networking.reflection.ReflectionClientProxy;
import pr1Java.server.Configuration;
import pr1Java.server.ConcurrentServer;
import pr1Java.services.reflection.ReflectionServices;

import java.net.Socket;

public class ReflectionConcurrentServer extends ConcurrentServer {

    private final ReflectionServices services;

    public ReflectionConcurrentServer(Integer port, ReflectionServices services) {
        super(port);
        this.services = services;
    }

    @Override
    protected Thread createClientProxyThread(Socket connection) {
        Configuration.logger.traceEntry("entering with", connection);

        ReflectionClientProxy clientProxy = new ReflectionClientProxy(services, connection);
        Thread clientProxyThread = new Thread(clientProxy);

        Configuration.logger.traceExit(clientProxyThread);
        return clientProxyThread;
    }
}

File: pr1-cs/Server/Reflection/ReflectionConcurrentServer.cs
using System.Net.Sockets;
using System.Threading;
using Networking.Reflection;
using Services.Reflection;

namespace Server.Reflection
{
    public class ReflectionConcurrentServer : ConcurrentServer
    {
        private readonly IServices _services;
        
        public ReflectionConcurrentServer(string host, int port, IServices services) : base(host, port)
        {
            _services = services;
        }

        protected override Thread CreateClientProxyThread(TcpClient connection)
        {
            var clientProxy = new RpcClientProxy(_services, connection);
            return new Thread(clientProxy.Run);
        }
    }
}